staging/src/k8s.io/apiserver/pkg/storage/feature/feature_support_checker.go

```diff
@@ -82,26 +82,60 @@ func (f *defaultFeatureSupportChecker) Supports(feature storage.Feature) (bool,
 func (f *defaultFeatureSupportChecker) CheckClient(ctx context.Context, c client, feature storage.Feature) error {
 	switch feature {
 	case storage.RequestWatchProgress:
-		return f.clientSupportsRequestWatchProgress(ctx, c)
+		return f.checkClient(ctx, c)
 	default:
 		return fmt.Errorf("feature %q is not implemented in DefaultFeatureSupportChecker", feature)
 
 	}
 }
 
-func (f *defaultFeatureSupportChecker) clientSupportsRequestWatchProgress(ctx context.Context, c client) error {
+func (f *defaultFeatureSupportChecker) checkClient(ctx context.Context, c client) error {
+	var wg sync.WaitGroup
+	errCh := make(chan error, len(c.Endpoints()))
+	retryCount := 3
+
+	retryFunc := func(ctx context.Context, c client, endpoint string, wg *sync.WaitGroup, errCh chan<- error) {
+		defer wg.Done()
+
+		for i := 0; i < retryCount; i++ {
```

Comment:
3 retries seems fairly low - etcd can definitely start visibly after apiserver in some setups, so it's risky.

I would rather use `BackoffUntil` with some exponential backoff, but timint out only after 10 minutes or sth.

Reply:
Thanks for the review, i will check that functionality and replace it.